### PR TITLE
Gh1325 scale over group max after metadata removal where group max lt clb max

### PR DESCRIPTION
**Pull request by sam-falvo**
While waiting for the PR for Gh1324 to land, which itself is blocked on Mimic supporting a feature at the moment, I might as well get early review on this PR too.

NOTE: This PR has code which is largely a duplicate of Gh1324, but not an exact copy.  This is a WIP PR at the moment, and I'll refactor before committing for sure.  If I can ask any reviewer(s) to not pollute the comments with observations on duplicated code, that'd be awesome.

### CODE REVIEW

**Comment by stretchservers** [comment]
Can one of the admins verify this patch?

File: otter/integration/tests/test_convergence.py [hunk 1]
```diff
@@ -67,6 +69,135 @@ def _check_fds(_):
...
+    def test_scale_over_group_max_after_metadata_removal_reduced_grp_max(self):
+        """
+        Attempt to scale over the group max, but only after metadata removal on
+        some random sampling of servers.  Note that this version exercises the
+        path that group max is less than CLB max.
+        """
+
+        rcs = TestResources()
+
+        self.untouchable_scaling_group = ScalingGroup(
```
Comment (cyli) [inline otter/integration/tests/test_convergence.py]: Apologies, what is the untouchable group for?  (there are no assertions about servers on this group)

Comment (sam-falvo) [inline otter/integration/tests/test_convergence.py]: The assertions are yet to be written.

Comment (sam-falvo) [inline otter/integration/tests/test_convergence.py]: The assertions will verify that what servers existed prior to this test remain there after the completion of the test.  I guess @kivattik had the idea of testing whether or not Convergence would "overflow" excess servers to another scaling group?

Comment (cyli) [inline otter/integration/tests/test_convergence.py]: Ok, thanks :D

**Comment by cyli** [comment]
LGTM

**Pull request merged**